src/reanimated2/component/ScrollView.tsx

```diff
@@ -25,27 +24,22 @@ const AnimatedScrollViewComponent = createAnimatedComponent(ScrollView);
 export const AnimatedScrollView = forwardRef(
   (props: AnimatedScrollViewProps, ref: ForwardedRef<AnimatedScrollView>) => {
     const { scrollViewOffset, ...restProps } = props;
-    const animatedRef = (
-      ref === null
-        ? // eslint-disable-next-line react-hooks/rules-of-hooks
-          useAnimatedRef<ScrollView>()
-        : ref
-    ) as AnimatedRef<AnimatedScrollView>;
-
-    if (scrollViewOffset) {
-      // eslint-disable-next-line react-hooks/rules-of-hooks
-      useScrollViewOffset(animatedRef, scrollViewOffset);
-    }
+    const animatedRef = useAnimatedRef<AnimatedScrollView>();
+    const outRef = ref ?? animatedRef;
+
+    // @ts-expect-error TODO: Allow null as `useScrollViewOffset` ref argument.
+    useScrollViewOffset(null, scrollViewOffset);
```

Comment:
cc @szydlovsky

Pls make it work in `useScrollViewOffset` (we are prisoners of our own API 💀)

Reply:
@tjzel do we really need nullable ref here that badly? Changing it will really complicate the hook implementation. The current native implementation also throws a warning when there is no ref - which I remember may have been your idea. After all, this hook with a null ref does essentially nothing so what do we need it for?

Reply:
Definitely we don't need to use nullable ref here. However:

1. We shouldn't call this hook conditionally like before.
2. The user can pass to `Animated.ScrollView` either:
    2.1 An AnimatedRef - then we have no problems whatsoever.
    2.2 A plain ref from React - then it will cause a runtime error.

In scenario 2.2 we would need to check if the passed ref is not an Animated one. If it's not, my idea would be to pass `null` to `useScrollViewOffset` hook as a special case that it shouldn't do anything. Keep in mind that we cannot just pass `animatedRef` to it as for now, since in this scenario it's never getting attached to anything and causes another error.

Reply:
@tjzel get yourself a new main 👍

Reply:
🥳